### PR TITLE
Thomaafl 20 api user table get users by name

### LINKED ISSUE #20: API: User table - GET - Get all users by name
## Functions to implement in domain/user.js:
- Add function findManyByLastName (basically a copy of findManyByLastName
- Add function findByName that uses the functions above to check if an input string is either a full name or only first/last, and then return the user(s).

## Endpoint - getAll
the parameter will be name, which can be either a first/lastname or a full name. The FindByName functon will take care of this. 
Method: GET
URL: users?name=Joe
Success:
Response code: 200
Response body type: array of User (s)
``` 
{
  "status": "success",
  "data": {
    "users": [
      {
        "id": 1,
        "cohort_id": 1,
        "role": "STUDENT",
        "firstName": "Joe",
        "lastName": "Bloggs",
        "email": "student@test.com",
        "biography": "Hello, world!",
        "githubUrl": "student1"
      }
    ]
  }
}
``` 

Error conditions:  
404 : Error: Not found
``` 
{
  "status": "fail",
  "message": "User not found"
}
```

### PR DESCRIPTION
**Pull request by thomaafl**
https://github.com/orgs/boolean-uk/projects/19/views/2?pane=issue&itemId=85014572&issue=boolean-uk%7Cteam-dev-server-ex-2410-team1%7C20

### CODE REVIEW

**Review by alihaiderkhannn — DISMISSED** [review]
Great code :)

**Review by JosteinRuen — DISMISSED** [review]
Great job with the filtering!

File: src/domain/user.js [hunk 1]
```diff
@@ -156,6 +157,41 @@ export default class User {
...
+    return User._findMany('lastName', lastName)
+  }
+
+  static async findByName(name) {
+    // Split the name into first and last name if it contains a space
+    let [firstName, lastName] = name.split(' ');
+    
+    // If it's a full name
+    if (lastName) {
+      console.log("firstName: ", firstName, "lastName: ", lastName);
```
Comment (allendewilzin) [inline src/domain/user.js]: Remove console.log()

File: src/domain/user.js [hunk 2]
```diff
@@ -156,6 +157,41 @@ export default class User {
...
+      let users = await User._findWithFullName({
+        firstName: firstName,
+        lastName: lastName
+      });
+      console.log("users: ", users);
+      if (users.length > 0) {
+          return users; 
+      }
+    }
+    console.log("only one name")
```
Comment (allendewilzin) [inline src/domain/user.js]: Remove console.log()

**Review by allendewilzin — CHANGES_REQUESTED** [review]

**Review by alihaiderkhannn — APPROVED** [review]

**Review by JosteinRuen — APPROVED** [review]
All console logs removed, awesome

**Comment by thomaafl** [comment]
@allendewilzin changed and fixed!

**Review by allendewilzin — APPROVED** [review]

**Pull request merged**